### PR TITLE
16480: Refactor Happy Mapper Import Code

### PR DESCRIPTION
**Pull request by saikumar9**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [16480](https://devops.dchbx.org/redmine/issues/16480) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | 3 |
| Peer Review Story Points | n/a |
| Ticket Author | @DFos85 |

### CODE REVIEW

File: app/models/financial_assistance/application.rb [hunk 1]
```diff
@@ -359,6 +359,123 @@ def next_incomplete_applicant
     end
   end
 
+  def build_or_update_tax_households_applicants_eligibility_determinations(verified_family, primary_person, active_verified_household)
```
Comment (psk86) [inline app/models/financial_assistance/application.rb]: call it `build_or_update_tax_households_and_applicants_and_eligibility_determinations` or just `build_or_update_tax_households`

File: app/models/financial_assistance/application.rb [hunk 2]
```diff
@@ -375,6 +492,17 @@ def attestation_terms_on_parent_living_out_of_home
     end
   end
 
+  def new_eligibility_determination(tax_household_id, verified_eligibility_determination, benchmark_plan_id, source)
+    eligibility_determinations.build(
+      e_pdc_id: verified_eligibility_determination.id,
+      benchmark_plan_id: benchmark_plan_id,
+      max_aptc: verified_eligibility_determination.maximum_aptc,
+      csr_percent_as_integer: verified_eligibility_determination.csr_percent,
```
Comment (psk86) [inline app/models/financial_assistance/application.rb]: update attribute : csr_eligibility_kind

File: app/models/financial_assistance/application.rb [hunk 3]
```diff
@@ -359,6 +359,123 @@ def next_incomplete_applicant
...
+            end
+          end
+          #Applicant/TaxHouseholdMember block end
+
+          #Eligibility determination start.
+          verified_eligibility_determination = vthh.eligibility_determinations.max_by(&:determination_date) #Finding the right Eligilbilty Determination
+
+          #TODO find the right source Curam/Haven.
+          source = "Haven"
+          new_eligibility_determination(tax_household.id, verified_eligibility_determination, benchmark_plan_id, source) #Creating Eligibility Determination
```
Comment (psk86) [inline app/models/financial_assistance/application.rb]: create_new_eligibility_determination

**Comment by saikumar9** [comment]
Old FAA code.

**Pull request closed without merging**